include/mega/filesystem.h

```diff
@@ -379,43 +453,43 @@ struct MEGA_API FileSystemAccess : public EventTrigger
     void normalize(string *) const;
 
     // generate local temporary file name
-    virtual void tmpnamelocal(string*) const = 0;
+    virtual void tmpnamelocal(LocalPath&) const = 0;
 
     // obtain local secondary name
-    virtual bool getsname(string*, string*) const = 0;
+    virtual bool getsname(LocalPath&, LocalPath&) const = 0;
 
     // rename file, overwrite target
-    virtual bool renamelocal(string*, string*, bool = true) = 0;
+    virtual bool renamelocal(LocalPath&, LocalPath&, bool = true) = 0;
 
     // copy file, overwrite target, set mtime
-    virtual bool copylocal(string*, string*, m_time_t) = 0;
+    virtual bool copylocal(LocalPath&, LocalPath&, m_time_t) = 0;
 
     // delete file
-    virtual bool unlinklocal(string*) = 0;
+    virtual bool unlinklocal(LocalPath&) = 0;
 
     // delete empty directory
-    virtual bool rmdirlocal(string*) = 0;
+    virtual bool rmdirlocal(LocalPath&) = 0;
```

Comment:
Yes, I totally agree and wanted to do it that way.  However, on windows we are prevented from doing that for now because they do actually edit the string.  Because it's storing a std::string with utf16 characters, the terminating null needs to be two bytes as well, but for std::string the terminating null is only one byte.  So the functions for windows append another null, and then remove it again before returning.    So, the parameter can't be const (or it would be a lie) for now.    I think the way to get there is to incorproate this this PR for now which gives us the big benefit of having the type introduced, and without changing too much for now - I thought making that change now would make the PR too large and difficult to review in one go.   But the next step can be to convert the internal string to std::wstring for windows, and fix up those windows functions using it, so that the parameter is not changed, and can be passed const.   Updating those windows specific functions will be a large number of lines changed too, which I've avoided for now - hopefully in review the change between old and new should show we are using the same string in the same way, just by accessing it inside the LocalPath wrapper class.    Sergio also asked about that in the prior review comment, and I've opened a ticket (16393) for that future improvement.